amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/holds/HoldsEntityResource.java

```diff
@@ -0,0 +1,211 @@
+import org.alfresco.rm.rest.api.impl.ApiNodesModelFactory;
+import org.alfresco.rm.rest.api.impl.FilePlanComponentsApiUtils;
+import org.alfresco.rm.rest.api.model.HoldDeletionReason;
+import org.alfresco.rm.rest.api.model.HoldModel;
+import org.alfresco.service.cmr.model.FileFolderService;
+import org.alfresco.service.cmr.model.FileInfo;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.transaction.TransactionService;
+import org.alfresco.util.ParameterCheck;
+import org.springframework.beans.factory.InitializingBean;
+
+/**
+ * Hold entity resource
+ *
+ * @author Damian Ujma
+ */
+@EntityResource(name = "holds", title = "Holds")
+public class HoldsEntityResource implements
+    EntityResourceAction.ReadById<HoldModel>,
+    EntityResourceAction.Update<HoldModel>,
+    EntityResourceAction.Delete,
+    InitializingBean
+{
+    private FilePlanComponentsApiUtils apiUtils;
+    private FileFolderService fileFolderService;
+    private ApiNodesModelFactory nodesModelFactory;
+    private HoldService holdService;
+    private TransactionService transactionService;
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        ParameterCheck.mandatory("nodesModelFactory", nodesModelFactory);
+        ParameterCheck.mandatory("apiUtils", apiUtils);
+        ParameterCheck.mandatory("fileFolderService", fileFolderService);
+        ParameterCheck.mandatory("holdService", holdService);
+        ParameterCheck.mandatory("transactionService", transactionService);
+    }
+
+    @Override
+    @WebApiDescription(title = "Get hold information", description = "Get information for a hold with id 'holdId'")
+    @WebApiParam(name = "holdId", title = "The hold id")
+    public HoldModel readById(String holdId, Parameters parameters)
+    {
+        checkNotBlank("holdId", holdId);
+        mandatory("parameters", parameters);
+
+        NodeRef hold = apiUtils.lookupAndValidateNodeType(holdId, RecordsManagementModel.TYPE_HOLD);
+        FileInfo info = fileFolderService.getFileInfo(hold);
+        return nodesModelFactory.createHoldModel(info);
+    }
+
+    @Override
+    @WebApiDescription(title = "Update a hold", description = "Updates a hold with id 'holdId'")
+    public HoldModel update(String holdId, HoldModel holdModel, Parameters parameters)
+    {
+        checkNotBlank("holdId", holdId);
+        mandatory("holdModel", holdModel);
+        mandatory("parameters", parameters);
+
+        NodeRef nodeRef = apiUtils.lookupAndValidateNodeType(holdId, RecordsManagementModel.TYPE_HOLD);
+
+        RetryingTransactionCallback<Void> callback = new RetryingTransactionCallback<Void>()
+        {
+            public Void execute()
+            {
+                if(holdModel.getName() != null)
+                {
+                    holdService.setHoldName(nodeRef, holdModel.getName());
+                }
+                if(holdModel.getDescription() != null)
```

Comment:
I think that if this is actually `null` it means to unset the description. Our PUT is always interpreted as setting the whole resource rather than a partial update.